
A stop word is a commonly used word (such as “the”, “a”, “an”, “in”) that a search engine has been programmed to ignore, both when indexing entries for searching and when retrieving them as the result of a search query.

In [1]:
import spacy

In [3]:
nlp = spacy.load('en_core_web_sm')

In [4]:
print(nlp.Defaults.stop_words)

{'yet', 'somewhere', 'would', 'ten', 'down', 'either', 'together', 'whereupon', 'whereafter', 'keep', 'now', 'who', 'please', 'mostly', 'about', 'has', 'by', 'all', 'until', 'however', 'everyone', '‘m', 'can', 'are', 'our', 'beyond', 'everywhere', '‘re', 'anything', 'onto', 'against', 'through', 'give', 'four', 'eight', 'over', 'thereby', 'often', 'name', 'i', 'to', 'amount', 'hence', 'used', 'anywhere', 'may', 'your', "'m", 'done', 'also', 'thereafter', 'if', 'thence', '’ll', 'few', 'mine', 'even', '’s', 'whither', 'my', 'beside', 'you', 'two', 'might', 'have', 'did', 'less', 'do', 'side', 'yourselves', 'unless', 'sometime', "'ve", 'indeed', 'least', 'since', 'none', "'s", 'as', 'than', 'into', 'herself', 'never', 'top', 'of', 'above', 'behind', 'n’t', 'someone', 'nowhere', 'around', 'in', 'out', 'which', 'once', 'enough', 'us', 'get', 'whereas', 'though', 'hers', 'but', 'only', 'twenty', 'put', 'the', 'part', 'being', 'she', 'these', 'using', 'thru', 'well', 'along', 'further', 'each

In [5]:
len(nlp.Defaults.stop_words)

326

**Method to check if a word is stop word or not**

In [6]:
nlp.vocab['is'].is_stop

True

In [7]:
nlp.vocab['AKGEC'].is_stop

False

In [8]:
nlp.vocab['Hello'].is_stop

False

To add stop words of your own

In [9]:
nlp.Defaults.stop_words.add('i.e')

In [11]:
nlp.vocab['i.e'].is_stop

True

In [12]:
len(nlp.Defaults.stop_words)

327

As we can see that at previously it was 326 and now after adding in defults it became 327

## let's see thow you would remove a stop word

In [13]:
nlp.vocab['done'].is_stop

True

In [14]:
nlp.vocab['done'].is_stop = False

In [15]:
nlp.vocab['done'].is_stop

False

we can see that we have remove from the stop category

**Removing the stop words**

In [16]:

s='''
Data science is the study of data. Like biological sciences is a study of biology, physical sciences, it’s the study of physical reactions. Data is real, data has real properties, and we need to study them if we’re going to work on them. Data Science involves data and some signs.

It is a process, not an event. It is the process of using data to understand too many different things, to understand the world. Let Suppose when you have a model or proposed explanation of a problem, and you try to validate that proposed explanation or model with your data.

It is the skill of unfolding the insights and trends that are hiding (or abstract) behind data. It’s when you translate data into a story. So use storytelling to generate insight. And with these insights, you can make strategic choices for a company or an institution.

We can also define data science as a field that is about processes and systems to extract data of various forms and from various resources whether the data is unstructured or structured.
The definition and the name came up in the 1980s and 1990s when some professors, IT Professionals, scientists were looking into the statistics curriculum, and they thought it would be better to call it data science and then later on data analytics derived.
'''

In [18]:
s=s.replace('\n','')
s = s.strip()
s

'Data science is the study of data. Like biological sciences is a study of biology, physical sciences, it’s the study of physical reactions. Data is real, data has real properties, and we need to study them if we’re going to work on them. Data Science involves data and some signs. It is a process, not an event. It is the process of using data to understand too many different things, to understand the world. Let Suppose when you have a model or proposed explanation of a problem, and you try to validate that proposed explanation or model with your data. It is the skill of unfolding the insights and trends that are hiding (or abstract) behind data. It’s when you translate data into a story. So use storytelling to generate insight. And with these insights, you can make strategic choices for a company or an institution. We can also define data science as a field that is about processes and systems to extract data of various forms and from various resources whether the data is unstructured o

In [19]:
s = nlp(s)

In [20]:
l=[]
for token in s:
  if token.is_stop:
    l.append(token)
print(l)


[is, the, of, is, a, of, it, ’s, the, of, is, has, and, we, to, them, if, we, ’re, to, on, them, and, some, It, is, a, not, an, It, is, the, of, using, to, too, many, to, the, when, you, have, a, or, of, a, and, you, to, that, or, with, your, It, is, the, of, the, and, that, are, or, behind, It, ’s, when, you, into, a, So, to, And, with, these, you, can, make, for, a, or, an, We, can, also, as, a, that, is, about, and, to, of, various, and, from, various, whether, the, is, or, The, and, the, name, up, in, the, and, when, some, IT, were, into, the, and, they, it, would, be, to, call, it, and, then, on]


As, i can see that the list of all stop word present in text

In [21]:
len(l)

125

125 stop words are present

## To get all the words which do not belong in the stop word category

In [22]:
tokens =[token for token in s if not token.is_stop]

In [23]:
print(tokens)

[Data, science, study, data, ., Like, biological, sciences, study, biology, ,, physical, sciences, ,, study, physical, reactions, ., Data, real, ,, data, real, properties, ,, need, study, going, work, ., Data, Science, involves, data, signs, ., process, ,, event, ., process, data, understand, different, things, ,, understand, world, ., Let, Suppose, model, proposed, explanation, problem, ,, try, validate, proposed, explanation, model, data, ., skill, unfolding, insights, trends, hiding, (, abstract, ), data, ., translate, data, story, ., use, storytelling, generate, insight, ., insights, ,, strategic, choices, company, institution, ., define, data, science, field, processes, systems, extract, data, forms, resources, data, unstructured, structured, ., definition, came, 1980s, 1990s, professors, ,, Professionals, ,, scientists, looking, statistics, curriculum, ,, thought, better, data, science, later, data, analytics, derived, .]


In [24]:
len(tokens)

125

Printing sentence without stopwords:

In [26]:
' '.join([token.text for token in s if not token.is_stop])

'Data science study data . Like biological sciences study biology , physical sciences , study physical reactions . Data real , data real properties , need study going work . Data Science involves data signs . process , event . process data understand different things , understand world . Let Suppose model proposed explanation problem , try validate proposed explanation model data . skill unfolding insights trends hiding ( abstract ) data . translate data story . use storytelling generate insight . insights , strategic choices company institution . define data science field processes systems extract data forms resources data unstructured structured . definition came 1980s 1990s professors , Professionals , scientists looking statistics curriculum , thought better data science later data analytics derived .'